### DISCUSSION TITLE
Upgrading existing project to Adonis v5

### DISCUSSION THREAD

**Discussion by emiliosanches** [adonisjs/core#2322] [Help]
Hello. I work on a project that uses Adonis (at the moment, v4.1).
But I saw that Adonis v5.0 supports Typescript and my team and I are interested on upgrading to v5.0 and starting to write code in typescript.
Can I upgrade an existing project written in Adonis 4 to Adonis 5? And if so, how could we do it?

**Comment by McSneaky** [comment]
There's no `node ace upgrade` or `adonis upgrade`
Have to start new project in V5 and then migrate codebase over manually. Some chunks you can copy-paste, some require rewrite. Depends on how complex your project is
If I recall correctly someone in Discord told he migrated 100-200 endpoint system (not small pet project) over to V5 in about a week

**Reply by emiliosanches** [reply]
It's a very big project. We have 6 micro services written on Adonis. Some of them have more endpoints than others. But I estimate there is  more than almost 200 endpoints or more than that.
Since it's a commercial project, I don't know if the PO would like us to stop the tasks already defined to upgrade the version of Adonis.

**Reply by thetutlage** [reply]
Then it is better to stick to 4.1. Coz, the upgrade path is not simple and will require a lot of manual labour.

**Reply by davebot9000** [reply]
It is my understanding that adonis v4 will no longer be getting security and bug fixes after v5 is fully released? If so, sticking with v4 doesn't sound advisable either. Or am I mistaken and missing something?

**Reply by mateus4k** [reply]
We could have an update guide showing all the points that need to be changed to migrate a project from v4.1 to v5. It would be quite useful 🙂

**Reply by McSneaky** [reply]
> We could have an update guide showing all the points that need to be changed to migrate a project from v4.1 to v5. It would be quite useful

There is :wink: 
https://docs.adonisjs.com/releases/out-of-preview#upgrading-to-the-latest-versions

**Reply by mateus4k** [reply]
Wow! I had already looked for it and I didn't find this page. Thank you very much!

**Reply by alannaidon** [reply]
> Wow! I had already looked for it and I didn't find this page. Thank you very much!

Hi. Did you managed to move from 4.1 to v5 without creating a new project and do everything manually?

I Followed the guide but a lot of things couldn't be solved eve with TS set to false under configs.

**Reply by McSneaky** [reply]
> > Wow! I had already looked for it and I didn't find this page. Thank you very much!
> 
> Hi. Did you managed to move from 4.1 to v5 without creating a new project and do everything manually?
> 
> I Followed the guide but a lot of things couldn't be solved eve with TS set to false under configs.

Aah, my bad. The link I shared is meant to be used when upgrading from preview version of V5 to stable version of V5

When migrating V4 project to V5 then easier to just create new project with `create-adonis-ts-app`

**Reply by hal2814** [reply]
Just to revisit this thread; AdonisJs v4 has been working fine but as it is 2022 now, it is my understanding v4 will no longer be receiving security updates and I will need to migrate my application to v5- I see a post [here](https://github.com/adonisjs/core/discussions/1485) asking for package.json data to help develop a guide for updating v4 applications to v5 from August of 2020. Has there been any progress on that guide? 

I see thetutlage saying the upgrade path will require a lot of manual labor, and McSneaky advising to use `create-adonis-ts-app` and manually migrating to a newly created project. 

I'm wondering if there is (or will be) a guide on what exactly needs to be copied over, and what doesn't, and how to go about it -or am I overthinking it?

**Reply by davebot9000** [reply]
Between the WIP status of the v5 documentation and the LOE to migrate I just decided to use Nest.js. If I was going to have to re-write everything might as well use a framework with a bit more momentum, better version support, better documentation behind it. They also have the added benefit of offering paid support (optional) and their documentation is pretty great too, plus their module system is exceptional and pretty well documented. (I get Adonis is a smaller project, not trying to be a negative -  just helpful) They also have out-of-the-box support for Typescript but also support JS too. 
Hated to abandon Adonis but simply did not have the time to wait nor am I am able to contribute to the project as I am just doing this for work and we are very small startup-type team.

I greatly appreciate all the efforts of the Adonis js team and really hope things workout and it keeps growing! You all have been great. Sorry again for any perceived negativity here! I genuinely enjoyed using your framework
